java/ql/test/library-tests/frameworks/apache-collections/Test.java

```diff
@@ -0,0 +1,990 @@
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;lazySortedMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			SortedMap out = null;
+			SortedMap in = (SortedMap)newTreeMapWithMapValue(source());
+			out = MapUtils.lazySortedMap(in, (Factory)null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.multiValueMap(in, (Factory)null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.multiValueMap(in, (Class)null);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.multiValueMap(in);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.multiValueMap(in, (Factory)null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.multiValueMap(in, (Class)null);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;multiValueMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			MultiValueMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.multiValueMap(in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;orderedMap;;;MapKey of Argument[0];MapKey of ReturnValue;value"
+			OrderedMap out = null;
+			Map in = (Map)newTreeMapWithMapKey(source());
+			out = MapUtils.orderedMap(in);
+			sink(getMapKey(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;orderedMap;;;MapValue of Argument[0];MapValue of ReturnValue;value"
+			OrderedMap out = null;
+			Map in = (Map)newTreeMapWithMapValue(source());
+			out = MapUtils.orderedMap(in);
+			sink(getMapValue(out)); // $hasValueFlow
+		}
+		{
+			// "org.apache.commons.collections4;MapUtils;true;populateMap;(Map,Iterable,Transformer);;Element of Argument[1];MapValue of Argument[0];value"
+			Map out = null;
+			Iterable in = (Iterable)newWithElement(source());
```

Comment:
Let's move the `(String)` cast here. And with the improved return type we don't need the cast to `Iterable`:
```suggestion
			Iterable in = newWithElement((String)source());
```